### PR TITLE
Joe/update GitHub workflows

### PR DESCRIPTION
**Pull request by joe-braley**
(Reopen from forked pull request from earlier)

Updates included:

1. Parameterizing the Dockerfiles found within.
2. Fixing minor bug in `validate-docker-images.sh`.
3. Adding secrets to the repo for where we reference our images.

Note on item # 3: Secrets should work for regular pull requests however they will not be passed to runners if the pull request comes from a forked pull request. To mitigate this, we can review any incoming PRs to be sure the code does nothing unexpected. Then reopen a pull request (not from the fork) to run any required checks. 

Please let me know if there are any questions or concerns.

### CODE REVIEW

File: .github/workflows/build-images.yml [hunk 1]
```diff
@@ -16,17 +16,39 @@ jobs:
       matrix:
         jdkversion: [11, 16, 17]
         baseimage: ["cbld", "mariner", "ubuntu"]
+        include:
```
Comment (karianna) [inline .github/workflows/build-images.yml:19]: I think this is missing the mariner 1.0 / 2.0 split as Bruno was mentioning

**Review by karianna — CHANGES_REQUESTED** [review]

File: .github/workflows/build-images.yml (hunk 1 above)
Comment (joe-braley) [reply .github/workflows/build-images.yml:19]: Will add!

Comment (joe-braley) [reply .github/workflows/build-images.yml:19]: Functionality added. Resolving.

**Comment by joe-braley** [comment]
@karianna @brunoborges This is ready for another once over. Should include support for Mainer 1.0 and 2.0. With 1.0 having the `cm-1` tag. Let me know if there are any issues please and thanks!

File: .devops/build.yml [hunk 2]
```diff
@@ -19,26 +23,50 @@ parameters:
         distro: cbld
         version: 11
         package: msopenjdk-11
+        image: "image-repository" # set to image repo on pipeline run
```
Comment (karianna) [inline .devops/build.yml:26]: I'm no YAML expert but could [Template Parameters](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runtime-parameters#:~:text=The%20parameters%20section%20in%20a%20YAML%20defines%20what,%7B%20%7B%20%7D%7D%20are%20replaced%20with%20parameter%20values) be used?

File: docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk [hunk 3]
```diff
@@ -0,0 +1,22 @@
+ARG IMAGE
+ARG TAG
+FROM ${IMAGE}:${TAG}
+
+MAINTAINER Microsoft OpenJDK Support <openjdk-support@microsoft.com>
+LABEL "Author"="Microsoft"
+LABEL "Support"="Microsoft OpenJDK Support <openjdk-support@microsoft.com>"
+
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
+ARG package=msopenjdk-16
```
Comment (karianna) [inline docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk:10]: Java 16 for a Java 11 file?

File: docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    tdnf -y upgrade && \
+    tdnf install -y tzdata ca-certificates freetype fontconfig && \
+    rm -rf /var/cache/tdnf && \
+    rpm -Uvh https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm && \
+    tdnf install -y mariner-repos-ui && \
+    tdnf install -y $package --nogpgcheck && \
+    echo java -Xshare:dump && \
+    java -Xshare:dump 
+
+ENV JAVA_HOME=/usr/lib/jvm/msopenjdk-16
```
Comment (karianna) [inline docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk:22]: same comment as above

File: docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+LABEL "Support"="Microsoft OpenJDK Support <openjdk-support@microsoft.com>"
+
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
+ARG package=msopenjdk-16
+
+RUN tdnf -y update && \
+    tdnf -y upgrade && \
+    tdnf install -y tzdata ca-certificates freetype fontconfig && \
+    rm -rf /var/cache/tdnf && \
+    rpm -Uvh https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm && \
```
Comment (karianna) [inline docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk:16]: is mariner CentOS 7 based?

File: docker/mariner-cm1/Dockerfile.msopenjdk-17-jdk [hunk 6]
```diff
@@ -0,0 +1,22 @@
+ARG IMAGE
+ARG TAG
+FROM ${IMAGE}:${TAG}
+
+MAINTAINER Microsoft OpenJDK Support <openjdk-support@microsoft.com>
+LABEL "Author"="Microsoft"
+LABEL "Support"="Microsoft OpenJDK Support <openjdk-support@microsoft.com>"
+
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
+ARG package=msopenjdk-16
```
Comment (karianna) [inline docker/mariner-cm1/Dockerfile.msopenjdk-17-jdk:10]: Should be 17

File: docker/mariner-cm1/Dockerfile.msopenjdk-17-jdk [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    tdnf -y upgrade && \
+    tdnf install -y tzdata ca-certificates freetype fontconfig && \
+    rm -rf /var/cache/tdnf && \
+    rpm -Uvh https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm && \
+    tdnf install -y mariner-repos-ui && \
+    tdnf install -y $package --nogpgcheck && \
+    echo java -Xshare:dump && \
+    java -Xshare:dump 
+
+ENV JAVA_HOME=/usr/lib/jvm/msopenjdk-16
```
Comment (karianna) [inline docker/mariner-cm1/Dockerfile.msopenjdk-17-jdk:22]: should be 17

File: validate-docker-images.sh [hunk 8]
```diff
@@ -80,7 +80,7 @@ done
 # LTS Versions only
 ubuntu_versions=("20.04" "18.04")
```
Comment (karianna) [inline validate-docker-images.sh:81]: 22.04?

**Review by karianna — CHANGES_REQUESTED** [review]

File: validate-docker-images.sh (hunk 8 above)
Comment (joe-braley) [reply validate-docker-images.sh]: Added 22.04

File: .devops/build.yml (hunk 2 above)
Comment (joe-braley) [reply .devops/build.yml:26]: I can look into using a template though the reason why these parameters are the way they are is to not have MCR references in the pipeline. We can set these when we run the pipeline in AzDO.

File: docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk (hunk 5 above)
Comment (joe-braley) [reply docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk:16]: @karianna @brunoborges question regarding the above comment. Infra team took a look at swapping the link in question out for [this one](https://packages.microsoft.com/cbl-mariner/2.0/prod/Microsoft/x86_64/). Though when we inspect the packages available we only see 11 available. We were thinking about a couple of potential solutions:

1. Change the resource we pull from for 11 leaving 17 as-is until the package is available.
2. Upload the packages to Mariner 2.0's feed (though we are unsure if we will need to wait until the coming release).

Thoughts?

Comment (milderhc) [reply docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk:16]: We update the link above with Mariner 2.0 repository for x64 binaries for the 11 Dockerfile.

For 17, as we commented in an email thread before. The binaries for 17 using the "recommends" feature weren't published in the last release, that is coming to the July PSU, and we will publish 17 to the Mariner 2.0 repository. So, for the 17 Dockerfile we have two options: 1. Remove it from this change. 2. Update the file with the link and avoid building 17 in CI.
I wouldn't recommend leaving CentOS.

File: .devops/build.yml (hunk 2 above)
Comment (karianna) [reply .devops/build.yml:26]: LMK if you have any luck else we can revisit this in a future PR.

File: docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk (hunk 5 above)
Comment (karianna) [reply docker/mariner-cm1/Dockerfile.msopenjdk-11-jdk:16]: I'm not quite clear if something needs to change here?

File: validate-docker-images.sh [hunk 9]
```diff
@@ -78,9 +78,9 @@ done
 
 # Validates existing LTS releases of Ubuntu
 # LTS Versions only
-ubuntu_versions=("20.04" "18.04")
+ubuntu_versions=("22.04" "20.04" "18.04")
```
Comment (karianna) [inline validate-docker-images.sh:81]: @brunoborges Adding 22.04 make sense to you?  Should we drop 18.04?

Comment (brunoborges) [reply validate-docker-images.sh:81]: We should keep 18.04 until its regular support ends next year.

File: .devops/build.yml (hunk 2 above)
Comment (joe-braley) [reply .devops/build.yml:26]: After experimenting with templates for a while I think there are more changes that would need to be made to get them to work correctly. I can make an item to explore doing this. We have more changes incoming to these dockerfiles and I would not want to block those with this PR.

**Review by milderhc — APPROVED** [review]

**Pull request merged**